telegram/_telegramobject.py

```diff
@@ -100,8 +103,26 @@ def _apply_api_kwargs(self) -> None:
             if getattr(self, key, True) is None:
                 setattr(self, key, self.api_kwargs.pop(key))
 
-    def __str__(self) -> str:
-        return str(self.to_dict())
+    def __repr__(self) -> str:
+        """
+        * `__repr__` goal is to be unambiguous
+        * `__str__` goal is to be readable
+        * `str()` calls `__repr__`, if `__str__` is not defined
+
+        In our case "unambiguous" and "readable" largely coincide, so we can use the same logic.
+
+        Prints the object in the form `ClassName(arg=value, arg=value, ...)`
```

Comment:
which advantage do you see in explicitly documenting `__repr__` and `__str__` over just stating their effect? I.e. we also only say "objects of this type are comparable in terms of equality` instead of documenting `__eq__` so far … 🤔

Reply:
1. Its technical documentation, so if a user wanted to for e.g. override methods, it would be clear what exactly is being overriden and what effect its causing

2. They define the behaviour of classes more clearly, specially for those who prefer that over a summary above.

Reply:
Fair enough. I'll document Str and repr here then. The other dunder methods can go into another pr or just doc-fixes